datafusion/src/physical_plan/repartition.rs

```diff
@@ -365,56 +382,98 @@ impl RepartitionExec {
 
         Ok(())
     }
+}
+
+#[derive(Debug)]
```

Comment:
The `AbortOnDrop` structure nicely encapsulates RAAI style, aborting a bunch of `JoinHandle`s on drop. What would you think about using that same structure in all of the operators that need this treatment (e.g. SortExec, etc) rather than using `PinnedDrop` and raw `JoinHandles` for the other 

You might even be able to encapsulate the JoinHandle creation into `AbortOnDrop`  so it was easier to ensure that all tasks spawned in physical plans were doing the right thing (aka we could avoid using `tokio::task::spawn` directly). Having a controlled wrapper around task spawning might be useful for other activities too (like being able to control which threadpool was used to run the task)